import "std";
import "sys";

func main() void {
    var a: std::result[[u16, void]] = std::result[[u16, void]]::init_value(0xBEEF);
    assert a.is_value();
    assert not a.is_error();
    sys::dump[[u16]](a.value());
    a.error(); # boom
}
################################################################################
# EF BE
# panic: attempted to retrieve error from std::result in the value state
